data/secscan_model/secscan_v4_model.py

```diff
@@ -202,11 +202,23 @@ def needs_reindexing_query(indexer_hash):
 
         # TODO(alecmerdler): We want to index newer manifests first, while backfilling older manifests...
         iterator = itertools.chain(
+            # Guarantees at least one batch of most recent manifests
             yield_random_entries(
                 not_indexed_query,
                 Manifest.id,
                 batch_size,
                 max_id,
+                # Because we have multiple workers and must stay random to ensure minimal duplicative work
+                # we will instead limit the set worked to the most recent batch.
+                max_id - batch_size,
+            ),
+            # Could be a huge batch of work
+            yield_random_entries(
+                not_indexed_query,
+                Manifest.id,
+                batch_size,
+                # compensation for the above batch guarentee
+                max_id - batch_size,
```

Comment:
Shouldn't this second iterator be left to `min_id ... max_id`. We still want to backfill the older manifests. Adding the iterator from `max_id - batch_size ... max_id` before this one should prioritize the new ones.

Reply:
The previous block was from `min_id -> max_id`  We are splitting it into two `min_id -> max_id - batch_size` and from `max_id - batch_size -> max_id` and the latter is the first one to be scanned